### PR TITLE
Improve html printing

### LINKED ISSUE #282: HTML printing improvements
- Add syntax highlighting maybe
- Change how whitespace is printed depending on whether containing is `<pre>` or display: block, etc
Ex: `"\n        <input type=\"checkbox\" name=\"test-checkbox\"> Test checkbox\n      "`

### PR DESCRIPTION
**Pull request by calebeby**
Closes #282

Now non-significant whitespace is collapsed when HTML elements are printed in error messages.
And HTML is syntax highlighted in error messages

Screen Shot 2021-11-26 at 6 30 38 AM

### CODE REVIEW

File: jest.setup.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// The PLEASANTEST_TESTING_ITSELF environment variable (used in utils.ts)
+// allows us to detect whether PT is running in the context of its own tests
+// We use it to disable syntax-highlighting in printed HTML in error messages
+// so that the snapshots are more readable.
+// When PT is used outside of its own tests, the environment variable will not be set,
+// so the error messages will ahve syntax-highlighted HTML
```
Comment (gerardo-rodriguez) [inline jest.setup.ts:6]:
```suggestion
// so the error messages will have syntax-highlighted HTML
```

File: src/utils.ts [hunk 2]
```diff
@@ -100,3 +101,12 @@ export const printStackLine = (
...
   return `    at ${location}`;
 };
+
+export const printColorsInErrorMessages =
+  // The PLEASANTEST_TESTING_ITSELF environment variable (set in jest.setup.ts)
+  // allows us to detect whether PT is running in the context of its own tests
+  // We use it to disable syntax-highlighting in printed HTML in error messages
+  // so that the snapshots are more readable.
+  // When PT is used outside of its own tests, the environment variable will not be set,
+  // so the error messages will ahve syntax-highlighted HTML
```
Comment (gerardo-rodriguez) [inline src/utils.ts:111]:
```suggestion
  // so the error messages will have syntax-highlighted HTML
```

File: src/serialize/index.ts [hunk 3]
```diff
@@ -66,46 +67,103 @@ export const deserialize = (input: string) =>
...
+          child = child.nextSibling;
+          consecutiveMergedText += child.textContent || '';
+        }
+        printedChildren.push(
+          whiteSpaceSetting === '' ||
+            whiteSpaceSetting === 'normal' ||
+            whiteSpaceSetting === 'nowrap' ||
+            whiteSpaceSetting === 'pre-line'
+            ? consecutiveMergedText.replace(
+                // Pre-line should collapse whitespace _except newlines
```
Comment (gerardo-rodriguez) [inline src/serialize/index.ts:106]: Typo?
```suggestion
                // Pre-line should collapse whitespace _except_ newlines
```

File: src/serialize/index.ts [hunk 4]
```diff
@@ -66,46 +67,103 @@ export const deserialize = (input: string) =>
...
   const selfClosing = el.childNodes.length === 0;
-  return `<${tagName}${
+  // We haver to tell kolorist to print the colors
+  // beacuse by default it won't since we are in the browser
+  // (the colored message gets sent to node to be printed)
+  colors.options.enabled = true;
+  colors.options.supportLevel = 1;
+
+  // Syntax highlighting groups
+  const hi = {
```
Comment (gerardo-rodriguez) [inline src/serialize/index.ts:138]: What does `hi` stand for? Can this be renamed to something more descriptive? 🤔

**Review by gerardo-rodriguez — APPROVED** [review]
This is awesome, @calebeby, so much better! 🎉 

I left a couple of small questions/suggestions. I approve pending resolution of those edits.

Let me know if you would like another review, otherwise, LGTM! 👍

File: src/serialize/index.ts (hunk 4 above)
Comment (calebeby) [reply src/serialize/index.ts:138]: It stands for highlight. I guess we can rename it to highlight :shrug:

**Pull request merged**